src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs

```diff
@@ -197,10 +197,20 @@ public static IHostBuilder ConfigureDefaults(this IHostBuilder builder, string[]
 
         internal static void ApplyDefaultHostConfiguration(IConfigurationBuilder hostConfigBuilder, string[]? args)
         {
-            hostConfigBuilder.AddInMemoryCollection(new[]
+            // If we're running anywhere other than C:\Windows\system32, we default to using the CWD for the ContentRoot.
+            // However, since many things like Windows services and MSIX
+
+            // In my testing, both Environment.CurrentDirectory and Environment.GetFolderPath(Environment.SpecialFolder.System) return the path without
+            // any trailing directory separator characters. I'm not even sure the casing can ever be different from these APIs, but I it makes sense to
+            // ignore case for Windows path comparisons given the file system is usually (always?) going to be case insensitive for the system path.
+            var cwd = Environment.CurrentDirectory;
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || !string.Equals(cwd, Environment.GetFolderPath(Environment.SpecialFolder.System), StringComparison.OrdinalIgnoreCase))
```

Comment:
This PR has now been updated so hosting never sets the CWD as the content root if it's C:\Windows\system32 even if you explicitly call `IHostBuilder.ConfigureDefaults` or you use a `HostApplicationBuilder` without `DisableDefaults`.

You can still manually set this as the `ContentRoot` if you want to, but this will need to be announced as a breaking change.

Reply:
What will the content root default to?

Reply:
Unless explicitly set to something out, the truly-default default 😆 content root for hosting is  `AppContext.BaseDirectory` which is what `UseWindowsService` used to set it to for you before this change.

Reply:
Great!